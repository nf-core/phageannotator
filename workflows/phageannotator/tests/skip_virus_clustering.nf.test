nextflow_workflow {

    name "Test workflow: PHAGEANNOTATOR"
    script "workflows/phageannotator/main.nf"
    workflow "PHAGEANNOTATOR"

    // Dependencies
    tag "SEQKIT_SEQ"
    tag "COVERM_CONTIG"
    tag "FASTA_VIRUS_CLASSIFICATION_GENOMAD"
    tag "FASTA_VIRUS_QUALITY_CHECKV"
    tag "CAT_CAT"
    tag "BOWTIE2_BUILD"
    tag "GENOMAD_ENDTOEND"
    tag "GUNZIP"
    tag "FASTQ_ALIGN_BOWTIE2"


    test("Parameters: skip_virus_clustering = true") {
        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        [
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz', checkIfExists:true),
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz', checkIfExists: true)
                        ]
                    ],
                    [
                        [ id:'test2' ],
                        [
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz', checkIfExists:true),
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz', checkIfExists: true)
                        ]
                    ]
                )
                input[1] = Channel.of(
                    [
                        [ id:'test' ],
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists:true)
                    ],
                    [
                        [ id:'test2' ],
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists:true)
                    ]
                )
                """
            }
            params {
                outdir                  = "$outputDir"
                skip_virus_clustering   = true
                // parameters to decrease sensitivity for test data
                mash_screen_min_score   = 0.01
                genomad_min_score       = 0.01
                genomad_max_fdr         = 1
                // speed up options since tools are fully tested in subworkflows
                genomad_disable_nn      = true
                genomad_sensitivity     = 0.1
                checkv_minimal_db       = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() },
                { assert path("${outputDir}/VirusAbundance/coverm/contig/all_samples_alignment_results.tsv").readLines().size() == 4 }
            )
        }
    }
}
