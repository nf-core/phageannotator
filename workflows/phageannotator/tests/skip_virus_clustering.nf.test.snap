{
    "Parameters: skip_virus_clustering = true": {
        "content": [
            {
                "0": [

                ],
                "1": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2_virus_summary.tsv:md5,ae6d66b39f6ea43628db461e42012027"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test_virus_summary.tsv:md5,ae6d66b39f6ea43628db461e42012027"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test2"
                        },
                        "quality_summary.tsv:md5,5f1b0ace5264c11b2f5e8b0decbf2f9e"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "quality_summary.tsv:md5,5f1b0ace5264c11b2f5e8b0decbf2f9e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ]
                ],
                "5": [

                ],
                "6": [

                ],
                "7": [

                ],
                "8": [

                ],
                "9": [
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,0c310e8a973612ff99e0df847fb656ea",
                    "versions.yml:md5,0c310e8a973612ff99e0df847fb656ea",
                    "versions.yml:md5,0e57d76cb0b6149bbb32aded3595de9b",
                    "versions.yml:md5,0e57d76cb0b6149bbb32aded3595de9b",
                    "versions.yml:md5,1439e948f383899aaccbf49506923b04",
                    "versions.yml:md5,374c63bf7c31c6c34acbf8d74e6c7dcc",
                    "versions.yml:md5,6323b255a9c1b9ed3e1a41aea5bfa2bb",
                    "versions.yml:md5,74b2068d24b17a2f4a097a3063da3dae",
                    "versions.yml:md5,7cb8c67c6ff9cf01f799d6730d5ebd15",
                    "versions.yml:md5,7cb8c67c6ff9cf01f799d6730d5ebd15",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,987d2e54a85311b917599dc41c90418c",
                    "versions.yml:md5,a856eefde99260259440d942c1d56ff4",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b"
                ],
                "anicluster_reps_fna_gz": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ]
                ],
                "filtered_viruses_fna_gz": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ]
                ],
                "host_predictions_tsv": [

                ],
                "instrain_gene_info": [

                ],
                "marker_taxonomy_tsv": [

                ],
                "pharokka_output_tsv": [

                ],
                "versions": [
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,0c310e8a973612ff99e0df847fb656ea",
                    "versions.yml:md5,0c310e8a973612ff99e0df847fb656ea",
                    "versions.yml:md5,0e57d76cb0b6149bbb32aded3595de9b",
                    "versions.yml:md5,0e57d76cb0b6149bbb32aded3595de9b",
                    "versions.yml:md5,1439e948f383899aaccbf49506923b04",
                    "versions.yml:md5,374c63bf7c31c6c34acbf8d74e6c7dcc",
                    "versions.yml:md5,6323b255a9c1b9ed3e1a41aea5bfa2bb",
                    "versions.yml:md5,74b2068d24b17a2f4a097a3063da3dae",
                    "versions.yml:md5,7cb8c67c6ff9cf01f799d6730d5ebd15",
                    "versions.yml:md5,7cb8c67c6ff9cf01f799d6730d5ebd15",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,987d2e54a85311b917599dc41c90418c",
                    "versions.yml:md5,a856eefde99260259440d942c1d56ff4",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b"
                ],
                "virus_classification_tsv": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2_virus_summary.tsv:md5,ae6d66b39f6ea43628db461e42012027"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test_virus_summary.tsv:md5,ae6d66b39f6ea43628db461e42012027"
                    ]
                ],
                "virus_enrichment_tsv": [

                ],
                "virus_quality_tsv": [
                    [
                        {
                            "id": "test2"
                        },
                        "quality_summary.tsv:md5,5f1b0ace5264c11b2f5e8b0decbf2f9e"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "quality_summary.tsv:md5,5f1b0ace5264c11b2f5e8b0decbf2f9e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-29T14:58:48.608848512"
    }
}
